<launch>
    <arg name="rviz" default="true"/>
    <arg name="enable_slam" default="true"/>
    <arg name="kalman_dead_reckoning" default="true"/>

    <!-- kalman 노드 실행 -->
    <!-- <rosparam command="load" file="$(find hero_slam)/config/kalman.yaml" ns=""/> -->
    <!-- <node pkg="hero_slam" name="kalman" type="kalman_node.py" output="screen"/> -->

    <!-- gt pose를 path로 변환하여 발행 -->
    <!-- <node pkg="hero_slam" name="pose_gt2path" type="odom2path.py"/> -->

    <!-- dead reckoning 노드 실행 -->
    <rosparam command="load" file="$(find hero_slam)/config/dead_reckoning.yaml" ns=""/>
    <node pkg="hero_slam" name="dead_reckoning" type="dead_reckoning_node.py" output="screen"/>


    <!-- feature extraction node 실행 -->
    <rosparam command="load" file="$(find hero_slam)/config/feature.yaml" ns=""/>
    <node pkg="hero_slam" name="feature_extraction" type="feature_extraction_node_sim.py" output="screen"/>

    <!-- start the SLAM node -->
    <node pkg="hero_slam" name="slam" type="slam_node.py" output="screen">
        <rosparam command="load" file="$(find hero_slam)/config/slam.yaml" subst_value="true"/>
        <param name="enable_slam" value="$(arg enable_slam)"/>
        <param name="save_fig" value="false"/>
    </node>

    <node pkg="tf" name="world_to_odom_tf_publisher" type="static_transform_publisher" args="0 0 0 0 0 0 world odom 100" />
    <node pkg="tf" name="map_to_world_tf_publisher" type="static_transform_publisher" args="0 0 0 0 0 0 world map 100"/>

    <node pkg="rviz" name="rviz" type="rviz" args="-d $(find hero_slam)/rviz/kalman_and_deadreckoning.rviz"/>

</launch>
